File: ios/PersistKit/Record.swift
import Foundation

public struct Record {
    typealias Encoder = PropertyListEncoder
    typealias Decoder = PropertyListDecoder

    let identifier: String
    let kind: String
    let flags: UInt32
    let content: Data

    init(identifier: String, kind: String, flags: UInt32, content: Data) {
        self.identifier = identifier
        self.kind = kind
        self.flags = flags
        self.content = content
    }
}

File: android/app/src/main/java/co/omise/persister/Record.kt
package co.omise.persister

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "records")
data class Record(
    @PrimaryKey @ColumnInfo(name = "_id") var identifier: String,
    var kind: String,
    var flags: Int,
    @ColumnInfo(typeAffinity = ColumnInfo.BLOB) var content: ByteArray
)